SingularityBase/src/main/java/com/hubspot/singularity/SingularityTaskIdHistory.java

```diff
@@ -15,17 +18,20 @@
   private final Optional<ExtendedTaskState> lastTaskState;
   private final Optional<String> runId;
 
+  private static final Comparator<SingularityTaskHistoryUpdate> comparator = new Comparator<SingularityTaskHistoryUpdate>() {
+    @Override
+    public int compare(SingularityTaskHistoryUpdate o1, SingularityTaskHistoryUpdate o2) {
+      return o1.getTaskState().compareTo(o2.getTaskState());
+    }
+  };
+
   public static SingularityTaskIdHistory fromTaskIdAndTaskAndUpdates(SingularityTaskId taskId, SingularityTask task, List<SingularityTaskHistoryUpdate> updates) {
     ExtendedTaskState lastTaskState = null;
     long updatedAt = taskId.getStartedAt();
 
     if (updates != null && !updates.isEmpty()) {
-      SingularityTaskHistoryUpdate lastUpdate = updates.get(0);
-      for (SingularityTaskHistoryUpdate update : updates.subList(1, updates.size())) {
-        if (update.getTimestamp() > lastUpdate.getTimestamp()) {
-          lastUpdate = update;
-        }
-      }
+      Collections.sort(updates, comparator);
+      SingularityTaskHistoryUpdate lastUpdate = Iterables.getLast(updates);
```

Comment:
Could consider using `Collections.max` here for a slight efficiency gain, also avoids mutating an argument (which is usually dicey code style IMO)